api/v1beta1/network_disruption.go

```diff
@@ -714,3 +835,74 @@ func (s *NetworkDisruptionSpec) UpdateHostsOnCloudDisruption(cloudManager clouds
 
 	return nil
 }
+
+func (s *NetworkDisruptionSpec) Explain() []string {
+	explanation := []string{""}
+	explanation = append(explanation, "spec.network will apply tc rules on every target, causing a failure on specific network traffic. You can filter what network "+
+		"traffic is affected, using network.hosts, network.services, or network.allowedHosts. If you specify none of those, all outgoing traffic will be impacted.")
+
+	if s.Drop != 0 {
+		explanation = append(explanation, fmt.Sprintf("\tnetwork.drop applies a packet drop of %d percent.", s.Drop))
+	}
+
+	if s.Corrupt != 0 {
+		explanation = append(explanation, fmt.Sprintf("\tnetwork.corrupt will corrupt %d percent of packets.", s.Corrupt))
+	}
+
+	if s.Delay != 0 {
+		explanation = append(explanation, fmt.Sprintf("\tnetwork.delay applies a delay of %d ms to all packets.", s.Delay))
+
+		if s.DelayJitter != 0 {
+			explanation = append(explanation, fmt.Sprintf("\tnetwork.delayJitter applies a jitter of up to %d ms to the delay value to normally distribute the delay.", s.DelayJitter))
+		}
+	}
+
+	if s.BandwidthLimit != 0 {
+		explanation = append(explanation, fmt.Sprintf("\tnetwork.bandwidthLimit applies a bandwidth limit of %d ms to the filtered connections.", s.BandwidthLimit))
+	}
+
+	if s.Cloud != nil {
+		explanation = append(explanation, "\tnetwork.cloud will apply filters so that the chosen network failures affects traffic between the target and the specified cloud services.")
+		explanation = append(explanation, "\tWe will use the cloud providers' published public ip ranges.")
+		explanation = append(explanation, s.Cloud.Explain()...)
+	}
+
+	if s.HTTP != nil {
+		explanation = append(explanation, "\tnetwork.http will apply filters so that the chosen network failure affects when the target makes any HTTP requests "+
+			"of the following methods to the following paths.")
+		explanation = append(explanation, "\tThis will _not_ work on https traffic, only http traffic.")
+		explanation = append(explanation, s.HTTP.Explain()...)
+	}
+
+	if len(s.Hosts) != 0 {
+		explanation = append(explanation, "\tnetwork.hosts will apply filters so that the chosen network failure affects the traffic between the target and the following groups:")
+		for _, host := range s.Hosts {
+			explanation = append(explanation, host.Explain())
+		}
+	}
+
+	if len(s.Services) != 0 {
+		explanation = append(explanation, "\tnetwork.services is a convenience feature for targeting kubernetes services in the _same kubernetes cluster_. The chaos-controller will handle "+
```

Comment:
👍

Reply:
Should we get rid of this?

In the end it does not seem that convenient and causes confusion IMO

Most people focus on using this instead of FQDN, and as we are multi-cluster, this is causing a gap in their ability to properly create "cluster agnostic" specs.

If we get rid of this feature (cluster specific feature), we can instead
- have a single way to define the spec, less cognitive overhead
- stop having question about why it does not work when the service is not on the same cluster?
- provide simple working out of the box example, that just works, always

Not saying we should get rid of this in the community edition of the chaos-controller, but in our internal documentation, IMO, we should stop advertising about this, it is counter productive

Reply:
we don't have much internal documentation to get rid of tbh...

Reply:
I would really like us to connect with Doggo maintainer so we can hijack the GitHub documentation for Datadog engineers and prevent them for doing mistakes

As we are redirecting them to this documentation, it would be ideal to fix this documentation for Datadog engineers, and avoid presenting them informations they can't use.

With Doggo, we should be able to hijack the github pages, and tailor the information for our internal use

Reply:
Claire is right that our only internal documentation on this feature suggests not using it. I am in support of removing it, though I see the benefit in situations where an fqdn can't be constructed (theoretically some community users could be in this situation).